### ISSUE TITLE
/!\ Call for collaborators /!\

### ISSUE THREAD

**Issue by Anahkiasen** [rocketeers/rocketeer#746]
As you may all have noticed I have a bit of open source fatigue and the little motivation I do have to work on my open source projects is rarely spent on Rocketeer because of the insane amount of work it implies.

This is the current situation:
- Rocketeer 2.2 was released a _while back_ and Rocketeer 3 was started a few weeks after that. While it still works, advances in the PHP ecosystem mean that the dependencies it is tied to now throw errors which is... pretty problematic as I can't remove those dependencies without breaking BC.
- Rocketeer 3 was a major refactor of the codebase and brings a good load of new features, including an improved plugin system, FTP deployments, filesystem handling (ie copy a file from a server to another or to S3/Dropbox/etc during deploy), task options, freeform configuration, and a good bunch of other stuff. As well as some more far fetched features that are still at the embryo stage at the moment (web UI, on-server assistant and easier CI integration, etc)

The issue is that as Rocketeer 3 is a refactor, merging fixes between the two branches is a complicated affair: it requires me double checking every change to either branch and see if it applies to the other, rewriting the test (as the test suite is also majorly different). Long story short I basically have two projects to work on rolled into one.

Now I still care for Rocketeer, and try to answer issues as much as possible, but you all need to understand that deployments being a messy subject, a lot of the issues that get reported here are unrelated to Rocketeer or specific to one user's setup. 

So the gist of it is I need some help, I need people willing to lend a hand on the project. Not necessarily on the code aspect but at least some people to help triage issues and help me differentiate what is an actual bug and what is a specificity of the user's setup. If you do want to help on the code, I'll be there every step of the way to guide your through the codebase and do code reviews and so on, this would be teamwork in every way so even if you're not necessarily experienced I'm fine with it. Everybody started somewhere.

There are already people like that, some have come and gone, some are still here. **If you recognize yourself in this or want to contribute to open source for the sake of helping, leave a message here**. We can talk more in detail on Discord or Gitter or whatever suits you. 

Thank you in advance

**Comment by introwit** [comment]
@Anahkiasen 2 honest confessions:

- Have never used Rocketeer 
- My learning resource from the start for servers/deployments & Envoy (since that too is a task runner) has been Chris Fidao's Servers for hackers.

If you are okay with the 2 points Maxime I am all ready to learn from you and help with the said things. My expertise in PHP might be low compared to you but would definitely try my best to learn & apply the things needed.

**Comment by codedge** [comment]
@Anahkiasen I'd be happy to help. I use Laravel very often and consider myself as "experienced" with it... and also with PHP, of course ;)

Is there any starting point or introduction how to start helping you?

**Comment by m1guelpf** [comment]
Would love to help!

**Comment by boyanyordanov** [comment]
We are using Rocketeer for our deployments and I would love to help.

**Comment by nilansanjaya** [comment]
@Anahkiasen We heavily depend on rocketeer for all our deployments. Would love to contribute to make things better.

**Comment by cmmata** [comment]
I have some years of experience in PHP but no much free time. I could help triaging issues and collaborating on the code when I have time enough.

By the way, how does developing in this project works? How can I see who's working on X feature / bug? Could be nice to have some instructions on how to collaborate in a wiki, maybe some tags like 'working on' / 'pending' in issues, or use Github's project system.

**Comment by Anahkiasen** [comment]
@cmmata Currently I'm using [Zenhub](https://www.zenhub.com/) for project management since the project is public it's free, but the issues are currently not all in their right pipelines and the pipelines are tailored to one developper but I'll reorganize them if/when some people join in to help

**Comment by boyanyordanov** [comment]
The projects feature here could help to bring all the information in one place. 

Aside from this issue what does one need to do to join or we should just start picking up issues and making PRs?

**Comment by cmmata** [comment]
@Anahkiasen I have registered in Zenhub, but looks like I don't see Rocketeer's project management because I'm not in the repo. How can I join?

**Comment by Anahkiasen** [comment]
Yeah sorry I haven't had time to go through the thread and add you guys as collaborators, need to cleanup the pipelines first and update a bit the CONTRIBUTING.md

**Comment by ABSAhmad** [comment]
Hey,

would love to help too.

**Comment by mehdichati** [comment]
Hi,

I am available to help.

**Comment by oliverquynh** [comment]
Hi, I really want to help to maintain this project.
